File: src/test/java/edu/hw2/Task3Test.java
package edu.hw2;

import edu.hw2.Task3.DefaultConnectionManager;
import edu.hw2.Task3.FaultyConnection;
import edu.hw2.Task3.FaultyConnectionManager;
import edu.hw2.Task3.PopularCommandExecutor;
import edu.hw2.Task3.interfaces.Connection;
import edu.hw2.Task3.interfaces.ConnectionManager;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;

public class Task3Test {

    @Test
    @DisplayName("DefaultConnectionManager should return Connection interface implementation instance")
    void check_DefaultConnectionManager_getConnection() {
        ConnectionManager connectionManager = new DefaultConnectionManager();
        assertThat(connectionManager.getConnection()).isInstanceOf(Connection.class);
    }

    @Test
    @DisplayName("FaultyConnectionManager should return FaultyConnection instance")
    void check_FaultyConnectionManager_getConnection() {
        ConnectionManager connectionManager = new FaultyConnectionManager();
        assertThat(connectionManager.getConnection()).isInstanceOf(FaultyConnection.class);
    }
}

File: src/test/java/edu/hw3/Task3Test.kt
package edu.hw3

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class Task3Test {
    val task = Task3()

    @Test
    @DisplayName("Method should find frequency of strings in list")
    fun check_strings() {
        val list = listOf("код", "код", "код", "bug")


        val frequencyDict = task.freqDict(list)


        assertThat(frequencyDict).isEqualTo(mapOf("код" to 3, "bug" to 1))
    }

    @Test
    @DisplayName("Method should find frequency of integers in list")
    fun check_int() {
        val list = listOf(1, 1, 2, 2)


        val frequencyDict = task.freqDict(list)


        assertThat(frequencyDict).isEqualTo(mapOf(1 to 2, 2 to 2))
    }
}
